src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs

```diff
@@ -98,7 +99,138 @@ public void InputFileResolver_InitializeShouldCrawlFilesRecursively()
+
+            var result = target.ResolveInput(Path.Combine(_filesystemRoot, "TestProject"), "");
+
+
+            result.TestProjectPath.ShouldBe(Path.Combine(_filesystemRoot, "TestProject"));
+            result.ProjectContents.GetAllFiles().Count().ShouldBe(3);
+        }
+
+        [Fact]
+        public void InputFileResolver_InitializeShouldThrowOnMissingSharedProject()
+        {
+            string sourceFile = File.ReadAllText(_currentDirectory + "/Initialisation/TestResources/ExampleSourceFile.cs");
+
+            string projectFile = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+    <PropertyGroup>
+        <TargetFramework>netcoreapp2.0</TargetFramework>
+        <IsPackable>false</IsPackable>
+    </PropertyGroup>
+
+    <ItemGroup>
+    </ItemGroup>
+               
+     <Import Project=""../SharedProject/Example.projitems"" Label=""Shared"" />
+
+    <ItemGroup>
+        <ProjectReference Include=""../ExampleProject/ExampleProject.csproj"" />
+    </ItemGroup>
+</Project>";
+
+            string testProjectFile = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+    <PropertyGroup>
+        <TargetFramework>netcoreapp2.0</TargetFramework>
+        <IsPackable>false</IsPackable>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <PackageReference Include=""Microsoft.NET.Test.Sdk"" Version = ""15.5.0"" />
+        <PackageReference Include=""xunit"" Version=""2.3.1"" />
+        <PackageReference Include=""xunit.runner.visualstudio"" Version=""2.3.1"" />
+        <DotNetCliToolReference Include=""dotnet-xunit"" Version=""2.3.1"" />
+    </ItemGroup>
+               
+    <ItemGroup>
+        <ProjectReference Include=""../ExampleProject/ExampleProject.csproj"" />
+    </ItemGroup>
+                
+</Project>";
+
+            var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>
+                {
+                    { Path.Combine(_filesystemRoot, "SharedProject", "Shared.cs"), new MockFileData(sourceFile)},
+                    { Path.Combine(_filesystemRoot, "ExampleProject", "ExampleProject.csproj"), new MockFileData(projectFile)},
+                    { Path.Combine(_filesystemRoot, "ExampleProject", "Recursive.cs"), new MockFileData(sourceFile)},
+                    { Path.Combine(_filesystemRoot, "ExampleProject", "OneFolderDeeper", "Recursive.cs"), new MockFileData(sourceFile)},
+                    { Path.Combine(_filesystemRoot, "TestProject", "TestProject.csproj"), new MockFileData(testProjectFile)},
+                    { Path.Combine(_filesystemRoot, "TestProject", "bin", "Debug", "netcoreapp2.0"), new MockFileData("Bytecode") },
+                    { Path.Combine(_filesystemRoot, "TestProject", "obj", "Release", "netcoreapp2.0"), new MockFileData("Bytecode") },
+                });
+
+            var target = new InputFileResolver(fileSystem);
+
+            Assert.Throws<FileNotFoundException>(() =>
+                target.ResolveInput(Path.Combine(_filesystemRoot, "TestProject"), ""));
+
```

Comment:
I'm missing a test case for multiple shared projects included.

Reply:
Done